sanic/server.py

```diff
@@ -295,6 +309,29 @@ def data_received(self, data):
         except Exception:
             logger.exception("protocol.data_received")
 
+    # -------------------------------------------- #
+    # Python 3.7+
+    # -------------------------------------------- #
+
+    def get_buffer(self, sizehint=-1):
+        return self._buffer
+
+    def buffer_updated(self, nbytes: int) -> None:
+        data = self._buffer[:nbytes]
+        self._time = current_time()
+        self.recv_buffer += data
+
+        if len(self.recv_buffer) > self.app.config.REQUEST_BUFFER_SIZE:
```

Comment:
After playing with this new code, I wonder if this line should be
```
if len(self.recv_buffer) >= self.app.config.REQUEST_BUFFER_SIZE
```

Heres one thing I observed:
- Set REQUEST_BUFFER_SIZE to 128 bytes.
- BufferedProtocol chooses `self._buffer` with size 128 (self.recv_buffer is unsized/unlimited).
- Send a request to the app with 384 byte request size
- buffer_updated is called with nbytes 128
- 128 bytes is writren to recv_buffer
- `if nbytes > REQUEST_BUFFER_SIZE` is False, because 128 is not greater than 128
- data_received is set, and http headers are read
- http protocol calls _receive_more(), tries to unpause the transport, but its not paused
- buffer_updated() is called again with nsize 128
- a new 128 bytes of data is written to self.recv_buffer
- len(self.recv_buffer) is now 256, so self.transport.pause_reading() is now executed.
- data_received is set, and further http headers are read
- http protocol calls _receive_more() again which _unpauses_ the transport.
- repeats until all of the request is read and parsed by the HTTP protocol

So I'm wondering, should the transport be paused every time the recv_buffer size matches REQUEST_BUFFER_SIZE, or only when its _greater_ than that?
What effect does pausing the transport even have?

Reply:
The point of pausing would be to allow for it to drain. It is more of an issue with streaming then sending complete responses. It is a protection from exhausting memory limits.

Reply:
With that said, you are probably correct that it should be `>=`.